config/kernel.m4

```diff
@@ -601,9 +601,27 @@ dnl #
 dnl # Used internally by ZFS_LINUX_TEST_{COMPILE,MODPOST}
 dnl #
 AC_DEFUN([ZFS_LINUX_COMPILE], [
+	AC_ARG_VAR([KERNEL_CC], [C compiler for
+		building kernel modules])
+	if test "x$ac_cv_env_KERNEL_CC_set" == "xset"; then
+		KERN_CC=CC=$KERNEL_CC
+	fi
+	AC_SUBST(KERN_CC)
+	AC_ARG_VAR([KERNEL_LD], [Linker for
+		building kernel modules])
+	if test "x$ac_cv_env_KERNEL_LD_set" == "xset"; then
+		KERN_LD=LD=$KERNEL_LD
+	fi
+	AC_SUBST(KERN_LD)
+	AC_ARG_VAR([KERNEL_LLVM], [Binary option to
+		build kernel modules with LLVM/CLANG toolchain])
+	if test "x$ac_cv_env_KERNEL_LLVM_set" == "xset"; then
+		KERN_LLVM=LLVM=$KERNEL_LLVM
+	fi
+	AC_SUBST(KERN_LLVM)
 	AC_TRY_COMMAND([
```

Comment:
I'd suggest removing checks above and
```
        AC_TRY_COMMAND([
            KBUILD_MODPOST_NOFINAL="$5" KBUILD_MODPOST_WARN="$6"
-           make modules -k -j$TEST_JOBS $KERN_CC $KERN_LD $KERN_LLVM -C $LINUX_OBJ $ARCH_UM
+           make modules -k -j$TEST_JOBS ${KERNEL_CC:+CC=$KERNEL_CC} -C $LINUX_OBJ $ARCH_UM
            M=$PWD/$1 >$1/build.log 2>&1])
```
It would get rid of 3 variables injection into Makefiles.